firestore/integration_test_internal/src/aggregate_query_test.cc

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "firebase/firestore.h"
```

Comment:
This test suite needs to include the entire public API of AggregateQuery, including its copy/move constructors and assignment operators, the Query() method, the is_valid(), method, operator==, and operator!=. There will be a lot of tests. Welcome to C++ :)

See the tests from https://github.com/firebase/firebase-cpp-sdk/blob/main/firestore/integration_test_internal/src/transaction_options_test.cc for an example. Those tests are even not thorough enough for my liking, as they should also test moving/copying from self and to and from "invalid" objects (objects where is_valid() returns false).

This comment also applies to aggregate_query_snapshot_test.cc